source/Cute/Commands/Content/ContentTranslateCommand.cs

```diff
@@ -211,37 +230,74 @@ await ProgressBars.Instance()
+                                            var tryCount = 3;
+                                            string? translatedText = null;
                                             translatedText = await translate(translator, defaultLocaleFieldValue, defaultLocale.Code, cuteLanguage);
-                                        }
-                                        if (string.IsNullOrEmpty(translatedText))
-                                        {
-                                            if(!failedEntryIds.ContainsKey(entryId))
+
+                                            while (tryCount > 0 && string.IsNullOrEmpty(translatedText))
                                             {
-                                                failedEntryIds[entryId] = new List<string>();
+                                                tryCount--;
                                             }
 
-                                            failedEntryIds[entryId].Add(targetLocaleFieldName);
+                                            return (translatedText, true);
                                         }
-                                        flatEntry[targetLocaleFieldName] = translatedText;
-                                        entryChanged = true;
-                                        taskTranslate.Description = $"{Emoji.Known.Robot} Translating ({symbols} symbols translated)";
+                                        catch (Exception ex)
+                                        {
+                                            _console.WriteAlert($"Error translating text from {defaultLocale.Code} to {targetLocale.Code} using {tService}. Error Message: {ex.Message}");
+                                            return (string.Empty, false);
+                                        }
+                                    });
+
+                                    taskMap.Add(targetLocaleFieldName, translateWrapper());
+
+                                    //if (string.IsNullOrEmpty(translatedText))
+                                    //{
+                                    //    if(!failedEntryIds.ContainsKey(entryId))
+                                    //    {
+                                    //        failedEntryIds[entryId] = new List<string>();
+                                    //    }
+
+                                    //    failedEntryIds[entryId].Add(targetLocaleFieldName);
+                                    //}
+                                    //flatEntry[targetLocaleFieldName] = translatedText;
+                                    //entryChanged = true;
+                                    //taskTranslate.Description = $"{Emoji.Known.Robot} Translating ({symbols} symbols translated)";
+                                }
+                                taskTranslate.Increment(1);
+                            }
+                            
+                            Task.WaitAll(taskMap.Values.ToArray());
```

Comment:
Task.WaitAll() blocks the calling thread until all tasks complete, potentially wasting resources
A more efficient approach would use await Task.WhenAll() which doesn't block